pkg/kp/pcstore/consul_store.go

```diff
@@ -113,6 +113,62 @@ func (s *consulStore) Create(
 	return pc, nil
 }
 
+func (s *consulStore) Update(
+	podID types.PodID,
+	availabilityZone fields.AvailabilityZone,
+	clusterName fields.ClusterName,
+	podSelector klabels.Selector,
+	annotations fields.Annotations,
+	session Session,
+	oldPC fields.PodCluster,
+	modifyIndex uint64,
+) (fields.PodCluster, error) {
+	newPC := fields.PodCluster{
```

Comment:
Why not just let the caller pass a `fields.PodCluster` as `newPC` that already has the necessary ID?